packages/drivers/local-driver/src/localDocumentDeltaConnection.ts

```diff
@@ -69,17 +69,15 @@ export class LocalDocumentDeltaConnection extends DocumentDeltaConnection {
         // We use a promise resolve to force a turn break given message processing is sync
         // eslint-disable-next-line @typescript-eslint/no-floating-promises
         Promise.resolve().then(() => {
-            this.submitManager.add("submitOp", messages);
-            this.submitManager.drain();
+            this.submitCore("submitOp", messages);
```

Comment:
We have this problem without your changes, but it will become worse - Kafka limit for a batch is 1M. So if payload exceeds that size, it fails. I think driver has to have some hard limit that is lower that results in critical (non-retriable failure) and runtime should have the warning of half of that size (even though I believe limit is not communicated to runtime). That said it would be also great to propagate that limit from service to driver to runtime, in similar way we propagate service policy RE summarization config. Please take a look (or open an issue to follow up), I can dig examples of how it's done in summarization space.

Reply:
https://github.com/microsoft/FluidFramework/issues/7246